Cython/Compiler/Code.py

```diff
@@ -705,10 +723,42 @@ def __init__(self, proto=None, impl=None, init=None, cleanup=None,
         self.proto_block = proto_block
         self.name = name
         self.file = file
+        self.export = export
+        self.shared_utility_functions = self.parse_export_functions(export) if export else []
 
         # cached for use in hash and eq
         self._parts_tuple = tuple(getattr(self, part, None) for part in self.code_parts)
 
+    def parse_export_functions(self, export_proto: str) -> list:
+
+        assert '//' not in export_proto and '/*' not in export_proto and '*/' not in export_proto, \
+            f'Export block must not contain comments:\n{export_proto.strip()}\n in file {self.file}'
+
+        parsed_protos = []
+        proto_regex=r'''
+            ^(?:static\s)                                     # `static` keyword
+            (?P<ret_type>[^;()]+[\s*])                        # return type + modifier with optional * - e.g.: int *, float, const str *, ...
+            (?P<func_name>\w+)\((?P<func_params>[^)]*)\)$     # function with params - e.g. foo(int, float, *PyObject)
+        '''
+
+        for proto in export_proto.split(';\n'):
+            proto = proto.strip().replace('\n', '')
+            proto = re.sub(r'\s+', ' ', proto)
+
+            if len(proto) == 0:
+                continue
+            matched = re.match(proto_regex, proto, re.VERBOSE)
+            assert matched is not None, \
+                f"Wrong format of function definition in export block \n{export_proto.strip()}\n in {self.file}"
```

Comment:
It seems better to restrict the error output to the failing line.
```suggestion
            assert matched is not None, \
                f"Wrong format of function definition in export block \n{proto!r}\n in {self.file}"
```